src/core/main.cpp

```diff
@@ -567,6 +567,57 @@ int main( int argc, char * * argv )
 		}
 	}
 
+	// Test file argument before continuing
+	QFileInfo fileInfoLoad( fileToLoad );
+	QFileInfo fileInfoImport( fileToImport );
+	if( !fileToLoad.isEmpty() )
+	{
+		if( !fileInfoLoad.exists() )
+		{
+			printf( "The file %s does not exist.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( fileInfoLoad.isDir() )
+		{
+			printf( "%s is a directory.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( !(	fileInfoLoad.suffix().toLower() == "mmp" ||
+				fileInfoLoad.suffix().toLower() == "mmpz" ||
+				fileInfoLoad.suffix().toLower() == "mpt" ) )
+		{
+			printf( "%s is not an LMMS project file.\n",
+					fileToLoad.toStdString().c_str() );
+			exit( 1 );
+		}
+	}
+	else if( !fileToImport.isEmpty() )
+	{
+		if( !fileInfoImport.exists() )
+		{
+			printf( "The file %s does not exist.\n",
+					fileToImport.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( fileInfoImport.isDir() )
+		{
+			printf( "%s is a directory.\n",
+					fileToImport.toStdString().c_str() );
+			exit( 1 );
+		}
+		else if( !(	fileInfoImport.suffix().toLower() == "mid"	||
+				fileInfoImport.suffix().toLower() == "midi"	||
+				fileInfoImport.suffix().toLower() == "rmi"	||
+				fileInfoImport.suffix().toLower() == "flp"	||
```

Comment:
Such a message is useful before dropping support, when the feature is deprecated. It is not needed now.